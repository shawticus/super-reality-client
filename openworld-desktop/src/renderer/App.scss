:root {
  --color-background: #242526;
  --color-background-hover: #464647;
  --color-text: #A8ABAF;
  --color-separator:#A8ABAF;
}

:global {
  body {
    display: flex;
    overflow: hidden;
    width: -webkit-fill-available;
    height: -webkit-fill-available;
    margin: 0px;
    background-color: var(--color-background);
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
  }
}

#app {
  width: 100%;
  height: auto;
  display: block;
}

a:focus, a:hover, a:visited, a:link, a:active {
  text-decoration: none;
}

.nav {
  display: flex;
  list-style: none;
  margin: 0 -0.75em 1.5em;
  padding: 0;
}

.nav li {
  margin: 0 0.75em;
}

.active {
  font-weight: bold;
}

.mainWrapper {
  color: var(--color-text);
  width: 100%;
  height: 100%;
}