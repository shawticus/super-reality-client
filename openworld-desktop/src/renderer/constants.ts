// eslint-disable-next-line import/prefer-default-export
export const API_URL = "http://localhost:8000/api/v1/";
