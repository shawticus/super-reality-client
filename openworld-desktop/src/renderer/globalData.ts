import { mockLessonData } from "../mocks";

const globalData = {
  collections: {} as Record<string, any>,
  subjects: {} as Record<string, any>,
  lessons: {
    "5f4d58471f990c3396043511": mockLessonData,
  } as Record<string, any>,
};

export default globalData;
