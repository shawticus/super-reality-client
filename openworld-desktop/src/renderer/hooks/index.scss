.popup-container {
  position: absolute;
  left: 0;
  top: 0;
  width: 100vw;
  height: 100vh;
  background-color: #00000085;
  display: flex;
  z-index: 4;

  .popup-box {
    display: flex;
    flex-direction: column;
    border-radius: 4px;
    min-width: 100px;
    min-height: 100px;
    margin: auto;
    background-color: var(--color-section);
    max-width: calc(100% - 16px);
    max-height: calc(100% - 64px);
  }
}