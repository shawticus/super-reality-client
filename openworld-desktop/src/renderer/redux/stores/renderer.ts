import { combineReducers } from "redux";
import { configureStore } from "@reduxjs/toolkit";
import authSlice from "../slices/authSlice";
import renderSlice from "../slices/renderSlice";
import createLessonSlice from "../slices/createLessonSlice";
import createCollectionSlice from "../slices/createCollectionSlice";
import createSubjectSlice from "../slices/createSubjectSlice";
import commonPropSlice from "../slices/commonPropSlice";
import userDataSlice from "../slices/userDataSlice";

const rootReducer = combineReducers({
  auth: authSlice.reducer,
  render: renderSlice.reducer,
  createLesson: createLessonSlice.reducer,
  createCollection: createCollectionSlice.reducer,
  createSubject: createSubjectSlice.reducer,
  commonProps: commonPropSlice.reducer,
  userData: userDataSlice.reducer,
});

const store = configureStore({
  reducer: rootReducer,
});

export default store;
export type AppState = ReturnType<typeof rootReducer>;
