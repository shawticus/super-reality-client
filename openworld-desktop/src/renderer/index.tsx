import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { HashRouter as Router } from "react-router-dom";
import store from "./redux/stores/renderer";
import App from "./App";

ReactDOM.render(
  <Provider store={store}>
    <Router>
      <App />
    </Router>
  </Provider>,
  document.getElementById("app")
);
