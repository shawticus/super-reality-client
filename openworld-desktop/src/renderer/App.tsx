import React, { useCallback } from "react";
import { Switch, Route } from "react-router-dom";
import { useDispatch, useSelector } from "react-redux";
import TopNav from "./components/top-nav";
import "./App.scss";
import Test from "./views/test";
import Discover from "./views/discover";
import Learn from "./views/learn";
import Teach from "./views/teach";
import TopSearch from "./components/top-search";
import reduxAction from "./redux/reduxAction";
import CreateLesson from "./components/create-lesson";
import { AppState } from "./redux/stores/renderer";
import Splash from "./views/splash";
import Flex from "./components/flex";

export default function App(): JSX.Element {
  const isAuthenticated = useSelector((state: AppState) => state.auth.isValid);
  // const dispatch = useDispatch();
  // const location = useLocation();
  const scrollRef = React.useRef<HTMLDivElement | null>(null);
  const dispatch = useDispatch();

  const handleScroll = useCallback((): void => {
    if (scrollRef.current) {
      const { scrollTop } = scrollRef.current;
      reduxAction(dispatch, { type: "SET_YSCROLL", arg: scrollTop });
    }
  }, [scrollRef, dispatch]);

  /*
  const _authenticateFromLocalStorage = useCallback(() => {
    reduxAction(dispatch, { type: "AUTH_PENDING", arg: false });
    return localForage
      .getItem<string>("com.gamegen.classroom.auth.token")
      .then((token) => {
        if (token) {
          return Axios.post("http://localhost:3000/api/v1/auth/verify", null, {
            headers: { Authorization: `Bearer ${token}` },
            timeout: 3000,
          })
            .then((_response) => Promise.resolve(token))
            .catch((error) => Promise.reject(error));
        } else {
          return Promise.reject();
        }
      })
      .then((token) => {
        reduxAction(dispatch, { type: "AUTH_SUCCESSFUL", arg: token });
        return Promise.resolve();
      })
      .catch((error) => {
        reduxAction(dispatch, { type: "AUTH_FAILED", arg: false });
        return Promise.reject(error);
      });
  }, [dispatch]);

  const signOut = useCallback(() => {
    reduxAction(dispatch, { type: "AUTH_INVALIDATED", arg: false });
    return localForage.removeItem("com.gamegen.classroom.auth.token");
  }, [dispatch]);

  const onSignOutClick = (event: any) => {
    event.preventDefault();
    signOut();
    history.push("/auth");
  };
  */

  return isAuthenticated ? (
    <>
      <TopNav />
      <TopSearch />
      <div onScroll={handleScroll} ref={scrollRef} className="content">
        <div style={{ height: "52px" }} />
        <Switch>
          <Route path="/create/lesson" component={CreateLesson} />
          <Route path="/test" component={Test} />
          <Route path="/discover" component={Discover} />
          <Route path="/learn" component={Learn} />
          <Route path="/teach" component={Teach} />
        </Switch>
      </div>
    </>
  ) : (
    <Switch>
      <Route path="/" component={Splash} />
    </Switch>
  );
}
// <ProtectedRoute path="/admin" authPath="/auth" component={Admin} />
