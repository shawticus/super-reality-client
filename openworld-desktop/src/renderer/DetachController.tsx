/* eslint-disable no-underscore-dangle */
import React from "react";
import "./App.scss";
import { useSelector } from "react-redux";
import { AppState } from "./redux/stores/renderer";
import ViewLesson from "./components/view-lesson";

export default function DetachController(): JSX.Element {
  const { detached } = useSelector((state: AppState) => state.commonProps);

  return (
    <div className="content-deatched">
      <div className="content-wrapper">
        {detached?.type == "LESSON_VIEW" ? (
          <ViewLesson id={detached.arg._id} />
        ) : (
          <></>
        )}
      </div>
    </div>
  );
}
