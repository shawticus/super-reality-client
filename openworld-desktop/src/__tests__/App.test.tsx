/* eslint-env jest */
import React from "react";
import renderer from "react-test-renderer";
import Teach from "../renderer/views/teach";

test("renders app", () => {
  const tree = renderer.create(<Teach />).toJSON();
  expect(tree).toMatchSnapshot();
});
