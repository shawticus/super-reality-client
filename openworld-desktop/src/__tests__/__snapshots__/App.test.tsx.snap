// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<div
  className="mid"
>
  <div
    className="lesson-title-container"
  >
    <div
      className="lesson-icon"
    />
    <div>
      <div
        className="lesson-title"
      >
        Insert Lesson
      </div>
    </div>
  </div>
  <div
    className="collapsible-box"
    onClick={[Function]}
  >
    <div
      className="icon-collapse open"
    >
      <svg
        fill="var(--color-background)"
        height="8px"
        width="12.4px"
      >
        drop.svg
      </svg>
    </div>
    <div
      style={
        Object {
          "marginLeft": "8px",
        }
      }
    >
      Place
    </div>
  </div>
  <div
    className="collapsible-content"
    style={
      Object {
        "height": "0px",
        "overflow": "inherit",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        className="inner"
      >
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Collection
            </div>
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Title"
              value=""
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Subject
            </div>
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Title"
              value=""
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Parent Lesson (optional)
            </div>
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Title"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="collapsible-box"
    onClick={[Function]}
  >
    <div
      className="icon-collapse open"
    >
      <svg
        fill="var(--color-background)"
        height="8px"
        width="12.4px"
      >
        drop.svg
      </svg>
    </div>
    <div
      style={
        Object {
          "marginLeft": "8px",
        }
      }
    >
      Lesson Description
    </div>
  </div>
  <div
    className="collapsible-content"
    style={
      Object {
        "height": "0px",
        "overflow": "inherit",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        className="inner desc-authoring-grid"
      >
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "icon",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Icon
            </div>
            <div
              className="insert-media-container"
              onClick={[Function]}
              style={
                Object {
                  "height": "32px",
                  "width": "32px",
                }
              }
            >
              <svg
                fill="var(--color-text)"
                height="30px"
                style={
                  Object {
                    "margin": "auto",
                  }
                }
                width="30px"
              >
                add.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "title",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Lesson Title
            </div>
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Title"
              value=""
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "purpose",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Purpose
            </div>
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Title"
              value=""
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "tags",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Lesson Tags
            </div>
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Title"
              value=""
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "media",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Example Media
            </div>
            <div
              className="insert-images-div"
            >
              <div
                className="insert-media-container"
                onClick={[Function]}
                style={
                  Object {
                    "height": "125px",
                    "width": "100%",
                  }
                }
              >
                <svg
                  fill="var(--color-text)"
                  height="30px"
                  style={
                    Object {
                      "margin": "auto",
                    }
                  }
                  width="30px"
                >
                  add.svg
                </svg>
              </div>
              <div
                className="insert-media-container"
                onClick={[Function]}
                style={
                  Object {
                    "height": "125px",
                    "width": "100%",
                  }
                }
              >
                <svg
                  fill="var(--color-text)"
                  height="30px"
                  style={
                    Object {
                      "margin": "auto",
                    }
                  }
                  width="30px"
                >
                  add.svg
                </svg>
              </div>
              <div
                className="insert-media-container"
                onClick={[Function]}
                style={
                  Object {
                    "height": "125px",
                    "width": "100%",
                  }
                }
              >
                <svg
                  fill="var(--color-text)"
                  height="30px"
                  style={
                    Object {
                      "margin": "auto",
                    }
                  }
                  width="30px"
                >
                  add.svg
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="collapsible-box"
    onClick={[Function]}
  >
    <div
      className="icon-collapse open"
    >
      <svg
        fill="var(--color-background)"
        height="8px"
        width="12.4px"
      >
        drop.svg
      </svg>
    </div>
    <div
      style={
        Object {
          "marginLeft": "8px",
        }
      }
    >
      Step Authoring
    </div>
  </div>
  <div
    className="collapsible-content"
    style={
      Object {
        "height": "0px",
        "overflow": "inherit",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div
        className="inner step-authoring-grid"
      >
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "step",
            }
          }
        >
          <div
            className="container-with-desc"
          >
            <div>
              Step
            </div>
            <input
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="Event name"
              value=""
            />
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "event",
            }
          }
        >
          <div
            className="container-with-desc select-div-container"
          >
            <div
              className="select-div-title"
            >
              Event
            </div>
            <div
              className="select-container undefined"
            >
              <button
                className="select-button undefined "
                onClick={[Function]}
              >
                Two
              </button>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "flexWrap": "wrap",
              "gridArea": "images",
              "justifyContent": "space-between",
            }
          }
        >
          <div
            className="insert-images-div"
          >
            <div
              className="insert-media-container"
              onClick={[Function]}
              style={
                Object {
                  "height": "125px",
                  "width": "100%",
                }
              }
            >
              <svg
                fill="var(--color-text)"
                height="30px"
                style={
                  Object {
                    "margin": "auto",
                  }
                }
                width="30px"
              >
                add.svg
              </svg>
            </div>
            <div
              className="insert-media-container"
              onClick={[Function]}
              style={
                Object {
                  "height": "125px",
                  "width": "100%",
                }
              }
            >
              <svg
                fill="var(--color-text)"
                height="30px"
                style={
                  Object {
                    "margin": "auto",
                  }
                }
                width="30px"
              >
                add.svg
              </svg>
            </div>
            <div
              className="insert-media-container"
              onClick={[Function]}
              style={
                Object {
                  "height": "125px",
                  "width": "100%",
                }
              }
            >
              <svg
                fill="var(--color-text)"
                height="30px"
                style={
                  Object {
                    "margin": "auto",
                  }
                }
                width="30px"
              >
                add.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "action",
            }
          }
        >
          <div
            className="container-with-desc select-div-container"
          >
            <div
              className="select-div-title"
            >
              Action
            </div>
            <div
              className="select-container undefined"
            >
              <button
                className="select-button undefined "
                onClick={[Function]}
              >
                Two
              </button>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "text",
            }
          }
        >
          <textarea
            onChange={[Function]}
            onKeyDown={[Function]}
            placeholder=""
            style={
              Object {
                "minHeight": "64px",
                "resize": "vertical",
              }
            }
            value=""
          />
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "gridArea": "next",
            }
          }
        >
          <div
            className="container-with-desc select-div-container"
          >
            <div
              className="select-div-title"
            >
              Next
            </div>
            <div
              className="select-container undefined"
            >
              <button
                className="select-button undefined "
                onClick={[Function]}
              >
                Two
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="collapsible-box"
    onClick={[Function]}
  >
    <div
      className="icon-collapse open"
    >
      <svg
        fill="var(--color-background)"
        height="8px"
        width="12.4px"
      >
        drop.svg
      </svg>
    </div>
    <div
      style={
        Object {
          "marginLeft": "8px",
        }
      }
    >
      Previous Steps
    </div>
  </div>
  <div
    className="collapsible-content"
    style={
      Object {
        "height": "0px",
        "overflow": "inherit",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <div>
        <div
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <div
              className="inner-search-container"
            >
              <input
                className="inner-input"
              />
              <div
                style={
                  Object {
                    "height": "20px",
                    "margin": "auto",
                    "width": "20px",
                  }
                }
              >
                <svg
                  fill="var(--color-text)"
                  height="20px"
                  width="20px"
                >
                  search.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            className="select-container dark"
            style={
              Object {
                "marginLeft": "8px",
                "width": "-webkit-fill-available",
              }
            }
          >
            <button
              className="select-button dark "
              onClick={[Function]}
            >
              Name
            </button>
          </div>
        </div>
        <div
          className="steps-container"
        >
          <div
            className="step-container"
          >
            <div
              className="step-icon"
            />
            <div
              className="step-title"
            >
              Awesomeness
            </div>
            <div
              className="step-rating"
            >
              99
            </div>
          </div>
          <div
            className="step-container"
          >
            <div
              className="step-icon"
            />
            <div
              className="step-title"
            >
              Juice
            </div>
            <div
              className="step-rating"
            >
              45
            </div>
          </div>
          <div
            className="step-container"
          >
            <div
              className="step-icon"
            />
            <div
              className="step-title"
            >
              Draw bolts
            </div>
            <div
              className="step-rating"
            >
              42
            </div>
          </div>
          <div
            className="step-container"
          >
            <div
              className="step-icon"
            />
            <div
              className="step-title"
            >
              Hit something
            </div>
            <div
              className="step-rating"
            >
              37
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
